### ISSUE TITLE
Mod Crash

### ISSUE THREAD

**Issue by 1235789546** [simsulla/nwomod#419]
When the mod NWO 1.06 is launching and reaches the "creating textures for quad 23" crashes to desktop with the message that Victoria 2 stopped working.
I'm using V2 version 3.03.
The mod is placed in the right place and i have already tried erasing the cache's files of the game and mod (that solved this type of issue with other mods) but now didn't work.
I played the NWO mod 1.02 and didn't have this problem at all.

Any solutions please?

**Comment by simsulla** [comment]
When you installed 1.06 did you erase the old NWO folder and NWO.mod file?

**Comment by 1235789546** [comment]
Yes. I erased it and put the NEW mod. Didn't work.
I tried MERGING the NEW mod with the old one already installed. Didn't work.
I UNINSTALLED the game (erasing ALL files/folders, etc.) and then installed it again. Then I installed the NEW mod. Didn't work.
I tried re-downloading it. Didn't work.
I tried erasing the files in the "cache" folders, which solved crash problems in the ULTIMATE mod. Didn't work.
With the version 1.02 ALL quad textures function perfectly. I don't know WHY or WHAT has the Quad texture n° 23 that causes the error... it is SO frustrating... I LOVE this mod...
Thanks for the quick reply... please try fo find me a solution...

**Comment by simsulla** [comment]
You have Vic2 3.03 and are able to play the base game.  Please do not merge mods. The NWO download is always a complete download. Merging may not work.

Else by your description you have what is needed to play the ame. Let's double check that all is corrct.
1. Can you start a vanilla Vic2 game and 3.03 is displayed in the load screen?
2. Did you remove the cache file (sorry for asking but your problem sounds like this is the error?). The cache folder is usually C:\Users\YOU\Documents\Paradox Interactive\Victoria II\NwO. Remove gfx,logs,map.
3. The NWO zip file you downloaded, what is its size? Look in the Windows explorer. It should be 454 MB (476,895,463 bytes). If it is not, re-download the mod zip.

**Comment by 1235789546** [comment]
Hi again!
1. Yes.
2. If by "cahe file" you mean all the files INSIDE the cache folder, yes. I removed all of them on one occasion (The cache folder of THE GAME, Victoria 2 3.03, not the mod). In those files are the problems, there is the Quad Texture 23 that causes the crash.
NOTE: With the mod ULTIMATE, if you erased ALL the cache files of the game (Victoria 2 3.03) that solved a crash issue that some computers had. I tried that same thing to solve this, but didn't work.
3. Its size is 454 MB. I downloaded it 2 times.
Any suggestions?

**Comment by simsulla** [comment]
On 2. Never mind the game files. The mod cache files are the ones to remove. This is the path C:\Users\YOU\Documents\Paradox Interactive\Victoria II\**NwO**. "YOU" is your windows user name. Remove the files in NwO.

**Comment by 1235789546** [comment]
Thanks! I will try it! And tell you how it goes...

**Comment by 1235789546** [comment]
I tried it... Didn't work!!
I put the NWO mod in the mod folder of the game directory and ran it... the crash.
So then i deleted the files inside the "NwO" folder in C:\Users\EL KAISER\Documents\Paradox Interactive\Victoria II and then i ran it again..., and AGAIN the crash when loadin texture for quad 23...
Any other suggestion please? I want to play it!!!!

**Comment by 1235789546** [comment]
Okay, i have now updated the game to the 3.04 version and the mod still doesn't work!!!
I tried ALL of the above, but ALWAYS crashes at "Creating Texture for Quad 23".
Please Answer! Otherwise, if this messages keep being ignored, i will assume you have forgotten about my issue (due to the arrival of new issues) and i will create another issue thread exatcly like this one.

**Comment by simsulla** [comment]
@1235789546 the simple fact is that the configuration you mention does work. That is why i ask you to double check that the config is right. On your PC there are additional things done or misinformation.

Please stick with 3.03. It is supported it works.

Let's confirm that your vanilla 3.03 Vic2 Heart of Darkness installation is correct. Do this;
1. Clear the vanilla cache files
2. When the game loads. What is the check sum? You see it in the lower right corner?

(btw: 1.0.x yes that is ok. For 1.06 it should be 1.06)

**Comment by 1235789546** [comment]
Okay. I'll try again. I hope it works...

**Comment by 1235789546** [comment]
Okay, i did what you said Step-by-step. The only difference is that i decided to install the game by defect in "C:/Program Files/Paradox Interactive" instead of my other usual directory, which is in other disk, the "D:".
So i installed the game and then put the mod inside the "mod" folder, then i erased the "gfx" files and map/cache files of the GAME. When i ran the mod, it managed to reach "Creating texture for Quad 2140", but it crashed again. After that i decided to erase the cache files of the MOD, but crashes again at texture 2140. I don't know what to do!!!

**Comment by simsulla** [comment]
Something is incorrect with your setup, so we need to ensure that your standard Vic2 game is correctly installed. So please answer my question. On a clean install, erased vanilla game cache files what is the **check sum** of Victoria 2 HOD 3.03?

**Comment by 1235789546** [comment]
Please excuse my ignorance. You want me to install the game, then erase the cahe files, and ran through the launcher?
And if i do all that, what am i supposed to search for? When Yuri Gagarin is in the background is when appears the "Creating Checksum" message. Again, what i'm looking for?

**Comment by simsulla** [comment]
I want you to confirm that your standard Victoria 2 3.03 Heart of Darkness installation is correct. You do that by giving me the checksum. You can see the checksum in the lower right corner of the star screen.
![image](https://cloud.githubusercontent.com/assets/10263809/15204826/469a441e-1812-11e6-9a8a-6db9545e32ed.png)

What is it?

**Comment by 1235789546** [comment]
I have two Victoria 2 games, obviously not installed at the same time:
This is the one i got from Paradox: "V2 v3.03 (XSNQ)"
This i got after the other through torrent: "V2 v3.03 (PTIM)"
As a note, with BOTH games the NWO mod version 1.02 functions perfectly, but with neither works NWO 1.06...

**Comment by simsulla** [comment]
Clear the cache before determining your checksum. The should be the same it it is a correct installation. For V2 v3.03 it is [NSPY](https://forum.paradoxplaza.com/forum/index.php?threads/v2-heart-of-darkness-3-03-patch-out.692912/).

@1235789546 I don't know what is wrong in your base installation. I am afraid I can't help you. I am very sorry.

**Comment by 1235789546** [comment]
Done!. I did what you said: In the case of the game downloaded from the Paradox Web Page, now my vanilla checksum is "V2 v3.03 (NSPY)", which is the correct one, right?
In the case of the game downloaded through torrent, i can't make the checksum look other than "V2 v3.03 (PTIM)", it doesn't matter if i erase or not the cache files, the checksum is ALWAYS PTIM.
So let's focus on the legal game. What do i do now?

**Comment by simsulla** [comment]
Don't know. Searched the paradox forum for "Creating texture for Quad 23" and all indicates installation error. 
I had some similar problems when I was adding new provinces. In the mod/NWO/map/default.map file one have to use a relative path 

> definitions  = "../mod/NWO/map/definition.csv"

So your mod has to be installed in mod/NWO and a relative path must be valid. 
**WARNING (this will f-up vanilla Vic2):** you can test in the GAME map/default.map set 

> definitions  = "mod/NWO/map/definition.csv"

(if it does not work revert the change)

More than that I am afraid I can't help you. I am very sorry.

**Comment by 1235789546** [comment]
All right. Can´t make it work. If i find a solution, i'll comment it here. But seems i'm going to have to settle with Darkest Hour NWO2 mod. What i don't get is this: If i have ALWAYS installed my game on the SAME way than before, and if i have ALWAYS install the mods in the SAME way as always, how is it possible that ALL the mods i have function perfectly EXCEPT the newest version of this one? It doesn't make sense that it is an installation error. It has to be something else...
Thanks Anyway for the help!

**Comment by simsulla** [comment]
(it hurt my hart not to resolve this - one last thing you could try is to revert the last map changes i did in 1.06)
Replace the NWO/map with this [positions.txt](https://raw.githubusercontent.com/simsulla/nwomod/6bc71c1142bd3e5acdb47df7eb0d3f127d6d92f0/NWO/map/positions.txt)
And replace NWO/map/provinces.bmp with [provinces.bmp.
](https://github.com/simsulla/nwomod/blob/6bc71c1142bd3e5acdb47df7eb0d3f127d6d92f0/NWO/map/provinces.bmp?raw=true)

**Comment by 1235789546** [comment]
I just have 1 doubt. How many chances are that, considering v 1.02 and ANY OTHER mod works fine in my game, that the next version of NWO (i assume you still develop this mod) will work in my game? Are there many, or scarce?

**Comment by simsulla** [comment]
Here you have my [quad 23](https://drive.google.com/file/d/0B0i_pAZe5kNUMzRwaXZJSm9FTUk/view?usp=sharing) and [24](https://drive.google.com/file/d/0B0i_pAZe5kNUTnl1c1M0RXJ1bW8/view?usp=sharing). Place them in C:\Users\YOUR_USER\Documents\Paradox Interactive\Victoria II\NwO\map\cache\quad_textures

**Comment by 1235789546** [comment]
It partially WORKED! Thanks for that idea!
I mean partially because now i ALWAYS reach without problem the "creating texture for quad 1895" message. So now i would need you to upload the quads 1895 and 1896. But to be honest, the IDEAL thing to do (i mean for me) would be to pass me ALL your quads, but i'm assuming that that would be the HELL of a work, right?
Just pass me those 2 quads i ask for. Then i will see if i reach the game or not.
Thanks VERY MUCH for the help!

**Comment by simsulla** [comment]
Here is a zip with all the [quad_textures](https://drive.google.com/open?id=0B0i_pAZe5kNURndtcGVVZkNYbzg).

**Comment by 1235789546** [comment]
So close. So far. When i tried putting them all in the correct foler, didn't work at all. Not even past the texture 23. I re-installed the game but now not even your first quads 23 and 24 work.
Could you do me a favor? I found a 1-page-thread in Paradox Interactive in which my problem (but with other mod) is discussd and solved. Sadly, i didn't understand what i have to do. Could you look at it? It speaks about the Texture Quad 23 crash and about the cache files...
http://oldforum.paradoxplaza.com/forum/showthread.php?500567-Quad-23-crash-to-desktop

**Comment by simsulla** [comment]
0. Clear the NWO cache.
1. Start and exit the core Vic2 game.
2. Start NWO (now you get your quad 23 crash).
3. Open map/cache/time file in notepad (or notepad++). Add and remove a space then save the file. 
![image](https://cloud.githubusercontent.com/assets/10263809/15310713/318e5ce2-1bf3-11e6-903d-22d81bce0ed1.png)
to
![image](https://cloud.githubusercontent.com/assets/10263809/15310730/4c625898-1bf3-11e6-8549-7e81ab83e206.png)
4. start NWO again..

**Comment by 1235789546** [comment]
I did what you said in the first three steps, but there is no "time" file in Documents/Paradox Interactive/Victoria II/NwO/map/cache. It NEVER exists. when i 1st extract the mod, obviously there is no NwO folder in documents. To make that folder appear, i have to run the mod at least ONCE. After step 2 y did Step 3 and the 23 quad crash occurs and the folder is created in Documents, but there is no "time" file... ¿?

**Comment by simsulla** [comment]
One of my game installation isn't installed in default either... so no. that is not the issue.
 Maybe my interpretation of http://oldforum.paradoxplaza.com/forum/showthread.php?500567-Quad-23-crash-to-desktop was wrong, and it is the vanlilla (..Victoria II > map > cache) time file you should give a new time stamp, as the one in NwO isn't created until successful launch. Test that...

**Comment by 1235789546** [comment]
I want you to answer me some basic questions please: You don't need to explain, just answer "yes" or "no". It's to see if I'm doing something wrong:
Let's assume you install your game in the default location: 
1: Once installed, is there SOME file in the "game installation directory/map/cache? Or the folder is empty?
2: When you run the game for the FIRST TIME, it crashes or something or it has no problem at all? It has the correct checksum in that 1st time?
3: If you installed the game to ONLY play with a mod, do you put the mod in the "mod" folder of the game directory? If so, do you do that BEFORE or AFTER running the game for the 1st time?
4: Once the mod is put in place, do you FIRST run the mod or BEFORE that you run the game? (please remember you just want to play the MOD)
5: Do you run the mod through the LAUNCHER? Or with some other method? IS THERE some OTHER method?
Thanks!

**Comment by simsulla** [comment]
1. There are files. Remove them - they are there to make the first launch of the game faster,
2. Correct checksum. No problem at all.
3. Doesn't matter. NWO has it own cache, does not care for the main game cache.
4. See 3.
5. Yes. launcher

**Comment by 1235789546** [comment]
So in resume no, i'm not doing something wrong. Now this:
Once the MOD has succesfully launched and you have played it, etc., a folder is created in "Documents/Paradox Interactive/Victoria 2" called "NwO":
1: Inside that folder, i assume the "cache" folder is FULL with the quad textures and all that stuff, right? Confirm.
2: You said that the game, once installed, has files on its folder "map/cache" (the ones to be removed). Does that happen with the MOD? I mean you extract the mod and a folder apperas named "NWO", is inside there a "cache" folder with files on it?
3: AFTER running the game, is Inside the "Game Directory/mod/NWO/map" a folder called "cache"? If so, is it still empty?
4: Is there any other way to run and play the mod other than the launcher?
Thanks!

**Comment by simsulla** [comment]
1. yes
2. No, not inside the mod/NWO.
3. NO "Game Directory/mod/NWO/map/cache"!!! What is that? 
4. Use the launcher.
5. make a launche that fails. Then place the quads in the folder, than make another one.
6. I am afraid I don't know what is wrong in your setup. There is a beta 1.07, you could test and download that and see of it works for you. Other than that I can't help you.

**Comment by 1235789546** [comment]
Okay. I'll try that beta 1.07 version. Where is the Link?

**Comment by simsulla** [comment]
https://github.com/simsulla/nwomod/archive/v1.07-beta.zip

**Comment by 1235789546** [comment]
Hello again! Thanks for the Link!
A question: Which files have you modified between NWO 1.02 and 1.06? I ask so i can put the NEW ones (1.06) replacing the OLD ones (1.02) in the 1.02 version of the mod to see if that works... Or you don't remember which files were modified?

**Comment by simsulla** [comment]
To many files.... Did 1.07 work?

**Comment by 1235789546** [comment]
No. Do Remember 1.02 worked perfectly and 1.06 had the "texture 23" crash? Well, this version has the "texture 1895" crash... and i have tried pretty much everything: Replace the quads with the quads you gave me. Replace ONLY the ones that caused the crash with the ones you gave me... but nothing nothing works. I managed to get as far as "creating texture for quad 2140" but never to reach the menu... i just don't get WHAT is SO different between the version 1.02 whose texture quads are created without a problem and those in the next versions are not...

**Comment by 1235789546** [comment]
Hello again! It's been almost a month. I got another idea. But i don't know if its possible for you to do it. I need you to upload for me a folder called "NwO" (the one that gets created in My Documents/Paradox Interactive/Victoria II) to put it in MY "my documents/paradox interactive/victoria II" directory. As i NEVER reach the menu in ANY version of NWO, i need that folder in its "complete form" (that is, after the mod reached the menu without problems). Could you PLEASE do that?
